inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -0,0 +1,336 @@
+#include <mkldnn_extension_utils.h>
+#include <limits>
+#include "ie_parallel.hpp"
+#include "common/cpu_memcpy.h"
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+MKLDNNPadNode::MKLDNNPadNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
+        : MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNPadNode::getSupportedDescriptors() {
+    auto* padLayer = dynamic_cast<PadLayer*>(getCnnLayer().get());
+    if (padLayer == nullptr)
+        THROW_IE_EXCEPTION << "Cannot convert Pad layer.";
+
+    padsBegin = padLayer->GetParamAsUInts("pads_begin");
+    padsEnd = padLayer->GetParamAsUInts("pads_end");
+
+    SizeVector srcDims = padLayer->insData[0].lock()->getTensorDesc().getDims();
+    SizeVector dstDims = padLayer->outData[0]->getTensorDesc().getDims();
+    if (srcDims.size() != dstDims.size() || padsBegin.size() != srcDims.size() || padsEnd.size() != srcDims.size())
+        THROW_IE_EXCEPTION << padLayer->name << " Incorrect number of input/output dimensions!";
+
+    std::string pad_mode = padLayer->GetParamAsString("pad_mode");
+    if (pad_mode == "constant") {
+        padMode = CONSTANT;
+        padValue = padLayer->GetParamAsFloat("pad_value", 0.f);
+    } else if (pad_mode == "edge") {
+        padMode = EDGE;
+    } else if (pad_mode == "reflect") {
+        padMode = REFLECT;
+        for (size_t i = 0; i < srcDims.size(); i++) {
+            if ((srcDims[i] - 1) < padsBegin[i] || (srcDims[i] - 1) < padsEnd[i])
+                THROW_IE_EXCEPTION << padLayer->name << " Incorrect padsBegin or padsEnd for 'reflect' pad mode";
+        }
+    } else if (pad_mode == "symmetric") {
+        padMode = SYMMETRIC;
+        for (size_t i = 0; i < srcDims.size(); i++) {
+            if (srcDims[i] < padsBegin[i] || srcDims[i] < padsEnd[i])
+                THROW_IE_EXCEPTION << padLayer->name << " Incorrect padsBegin or padsEnd for 'symmetric' pad mode";
+        }
+    } else {
+        THROW_IE_EXCEPTION << padLayer->name
+                           << " Incorrect pad_mode. Only constants|edge|reflect|symmetric modes are supported!";
+    }
+
+    if (getParentEdges().size() != 1)
+        THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
+    if (getChildEdges().empty())
+        THROW_IE_EXCEPTION << "Incorrect number of output edges for layer " << getName();
+}
+
+void MKLDNNPadNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    InferenceEngine::Precision precision = getCnnLayer()->insData[0].lock()->getPrecision();
+    if (precision != InferenceEngine::Precision::FP32)
+        precision = InferenceEngine::Precision::FP32;
+    auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
+
+    auto srcDims = getParentEdgeAt(0)->getDims();
+
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = true;
```

Comment:
Let's support dynamic batch if `padsBegin[0] == 0` and `padsEnd[0] == 0`. 
Then, at the execution stage, you can get the actual batch using `batchToProcess()`.
It seems like it will be enough to correct `srcDims` and `dstDims`.
For example, you can see how it is used in Permute layer.
@dmitry-gorokhov сorrect please if I am wrong about anything.

Reply:
Support will be added in another PR